error.go

```diff
@@ -271,3 +263,17 @@ func joinStringsIfNonEmpty(delimiter string, parts ...string) string {
 		return strings.Join(filteredParts, delimiter)
 	}
 }
+
+func (e *Error) GetAllPrintableProperties() map[string]interface{} {
```

Comment:
I thought about it and it all depends on the perspective of how this method should be used. If it's just to show the data - maybe. But it would greatly complicate the method with no obvious reason. As this method would be, IMO, used only internally by the developer to either create some sort of representation of the error (like JSON) *or* to log the values of properties, I do not see the point of casting them to string. 

On top of that `interface{}` is quite gently covered by MarshalJSON and similar methods.

Reply:
I'm not sure I fully understood all of your arguments. Presently, I disagree on this point. Is the goal of this method is just to expose the same thing that is being added to error message in a more structured way, I fail to see how a string representation is not enough. 
On the other hand, returning a raw value of (otherwise possibly inaccessible) property may open the door for all sorts of illicit uses, and I'm not sure it is even worth the effort of carefully considering them all. In such a general purpose library as this, it is best to keep the API as clean as possible, and I would like to avoid adding new ways to access inner data without a very good reason to do so.